toolchest_client/tools/tool.py

```diff
@@ -342,6 +334,24 @@ def _wait_for_threads_to_finish(self, check_health=True):
         for thread in self.query_threads:
             thread.join()
 
+    def _wait_for_threads_to_upload(self):
+        """Waits for all jobs to finish uploading. To be used only at the start of a run."""
+        while True:
```

Comment:
`while True` is almost always anti-pattern – works, but worth flagging

Reply:
Fair, that's what I had thought -- we can refactor this, I just wanted to avoid the extra `time.sleep(5)` call, which can be noticeable if the file is small